api/bin/chainflip-ingress-egress-tracker/src/witnessing/state_chain.rs

```diff
@@ -122,95 +185,256 @@ impl Storable for WitnessInformation {
 	}
 }
 
-type DepositInfo<T> = (DepositWitness<T>, <T as Chain>::ChainBlockNumber, NetworkEnvironment);
-
-impl From<DepositInfo<Ethereum>> for WitnessInformation {
-	fn from((value, height, _): DepositInfo<Ethereum>) -> Self {
-		Self::Deposit {
-			deposit_chain_block_height: height,
-			deposit_address: hex_encode_bytes(value.deposit_address.as_bytes()),
-			amount: value.amount.into(),
-			asset: value.asset.into(),
-			deposit_details: value
-				.deposit_details
-				.tx_hashes
-				.map(|tx_hashes| DepositDetails::Ethereum { tx_hashes }),
-		}
+trait IntoDepositDetailsAnyChain {
+	fn into_any_chain(self) -> Option<DepositDetails>;
+}
+
+impl IntoDepositDetailsAnyChain for cf_chains::evm::DepositDetails {
+	fn into_any_chain(self) -> Option<DepositDetails> {
+		self.tx_hashes.map(|tx_hashes| DepositDetails::Ethereum { tx_hashes })
+	}
+}
+impl IntoDepositDetailsAnyChain for cf_chains::btc::Utxo {
+	fn into_any_chain(self) -> Option<DepositDetails> {
+		Some(DepositDetails::Bitcoin { tx_id: self.id.tx_id, vout: self.id.vout })
+	}
+}
+impl IntoDepositDetailsAnyChain for u32 {
+	fn into_any_chain(self) -> Option<DepositDetails> {
+		Some(DepositDetails::Polkadot { extrinsic_index: self })
 	}
 }
+impl IntoDepositDetailsAnyChain for Vec<H256> {
+	fn into_any_chain(self) -> Option<DepositDetails> {
+		Some(DepositDetails::Arbitrum { tx_hashes: self })
+	}
+}
+#[async_trait]
+trait DepositIntoWitnessInformation<C: Chain> {
+	async fn into_witness_information<StateChainClient>(
+		self,
+		height: <C as Chain>::ChainBlockNumber,
+		network: NetworkEnvironment,
+		state_chain_client: Arc<StateChainClient>,
+	) -> anyhow::Result<WitnessInformation>
+	where
+		StateChainClient: StorageApi + ChainApi + TrackerApi + 'static + Send + Sync;
+}
```

Comment:
This looks like it can be simplified - no need for async or sc client? (Or result??)

```suggestion
trait DepositIntoWitnessInformation<C: Chain> {
	async fn into_witness_information<StateChainClient>(
		self,
		height: <C as Chain>::ChainBlockNumber,
		network: NetworkEnvironment,
	) -> WitnessInformation;
}
```

Reply:
(Also, not sure we really need a macro)

Reply:
Actually - this is pretty confusing - we have two macros for implementing the same trait 😅